### ISSUE TITLE
ambient cache tiles cyclical eviction while offline

### ISSUE THREAD

**Issue by jjblum** [mapbox/mapbox-gl-native#11996]
**Platform:** Android
**Mapbox SDK version:** 5.5.1

### Steps to trigger behavior

 1. Ambient cache maps until you reach 50 MB limit before eviction.
 2. View ambient cache tiles while offline. EDIT - device may be connected to a WLAN without internet access
 3. Upon observing eviction, change zoom level, observe tiles reappear then be re-evicted (see Actual Behavior below)
 4. View ambient cache tiles while online - behavior is not observed

### Expected behavior

Ambient cache tiles are either available and are displayed normally (they have not yet been evicted) or unavailable and will not be displayed (evicted while device was last online, i.e. ambient caching new tiles that replace unused ones).

### Actual behavior

- The ambient cache tiles display as if they are available, but then seem to "unload" despite device being offline.

- Details disappear as if you zoomed out despite zoom level not changing. Most likely because the next zoom level tile in the pyramid is now displayed instead.

- Process continues until no tiles are displayed - about twice per second the detail level drops, eventually only the default image is shown, as if no ambient cache ever occurred. 

- Changing the zoom level will force the appropriate tile to appear briefly again, even if this unloading process has run its course. It is as if the tile is still cached but is being actively unloaded right after it is displayed.

- Behavior continues no matter what location is used after behavior begins, even locations where the behavior has already taken place.

- User must uninstall and reinstall application (clearing ambient cache completely) to stop behavior

In summary, I ask the following questions:

 1. What should eviction look like to a user viewing the tiles online vs. viewing ambient cache tiles offline?
 2. Should eviction occur while offline?
 3. Is there a way to warn the user that eviction has started to occur? 
 4. Does the API provide a way to check ambient cache size? 
 5. Can the OfflineRegion API be used with ambient cache?

EDIT 2018-5-25: update steps to trigger
